spinnaker_graph_front_end/examples/hello_world_untimed/hello_world_vertex.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (c) 2017-2019 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from enum import Enum
+import logging
+from spinn_utilities.overrides import overrides
+from pacman.model.graphs.machine import MachineVertex
+from pacman.model.resources import ResourceContainer, VariableSDRAM
+from spinn_front_end_common.utilities.constants import (
+    SYSTEM_BYTES_REQUIREMENT, BYTES_PER_WORD)
+from spinn_front_end_common.utilities.helpful_functions import (
+    locate_memory_region_for_placement)
+from spinn_front_end_common.abstract_models import (
+    AbstractGeneratesDataSpecification)
+from spinn_front_end_common.interface.buffer_management.buffer_models import (
+    AbstractReceiveBuffersToHost)
+from spinn_front_end_common.interface.buffer_management import (
+    recording_utilities)
+from spinnaker_graph_front_end.utilities import SimulatorVertex
+from spinnaker_graph_front_end.utilities.data_utils import (
+    generate_system_data_region)
+import numpy
+from pacman.executor.injection_decorator import inject_items
+
+logger = logging.getLogger(__name__)
+
+
+class HelloWorldVertex(
+        SimulatorVertex, AbstractGeneratesDataSpecification,
+        AbstractReceiveBuffersToHost):
+
+    DATA_REGIONS = Enum(
+        value="DATA_REGIONS",
+        names=[('SYSTEM', 0),
+               ('PARAMS', 1),
+               ('STRING_DATA', 2)])
+
+    PARAMS_BASE_SIZE = BYTES_PER_WORD * 2
+
+    def __init__(self, n_repeats, label, constraints=None):
+        super(HelloWorldVertex, self).__init__(
+            label, "hello_world.aplx", constraints=constraints)
+
+        self._text = label
+        text_extra = len(label) % BYTES_PER_WORD
+        if text_extra != 0:
+            for _ in range(BYTES_PER_WORD - text_extra):
+                self._text += ' '
+
+        self._n_repeats = n_repeats
+
+    @property
+    @overrides(MachineVertex.resources_required)
+    def resources_required(self):
+        resources = ResourceContainer(
+            sdram=VariableSDRAM(
```

Comment:
this is now allocating a ton more sdram for a given "runtime" than you need. yuck. if run steps != to max_index, and more importantly if run steps was significantly bigger than max_index, you'll allocate, or at least attempt to allocate, more sdram than you need. 

For example run_until_complete(1 billion).
n_repeats = 6. 

Id predict this would fail in res allocating. as 1 billion bytes doth not exist on a chip.

Reply:
I can see how you were confused by this, but it was really by design (I wanted it to work like that).  Note that the memory reserved will always be the same, as it will be the amount that allows the maximum number of steps to run.  I have now removed this option to remove the confusion in any case.

Reply:
I should add that run_until_complete(1 billion) would have worked, as it would have used auto-pause-and-resume in that case, which should work just the same for step-based simulation as it does for timed-based simulations but it will break the run down into steps rather than times.  Note that time steps are just steps which are forcefully delayed.

Reply:
ok. but if id deactivated auto pause and resume? it should still have been able to run if it allocated just enough sdram as it needed. anyhow. will review changes shortly

Reply:
If you deactivate auto-pause-and-resume then yes, it would have failed where it should have run.  The changed version will now correctly not run!